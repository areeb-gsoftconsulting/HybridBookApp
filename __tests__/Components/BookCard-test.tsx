import React from 'react';
import {fireEvent,render} from '@testing-library/react-native';
import BookCard from '../../src/components/BookCard';


describe('Component testing', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  
  test('should render books component', () => {
    const bookCard = render(<BookCard />).toJSON();
    expect(bookCard).toMatchSnapshot();
  });
});
