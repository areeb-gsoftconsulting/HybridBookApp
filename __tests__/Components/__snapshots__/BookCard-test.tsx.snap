// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component testing should render books component 1`] = `
<RCTScrollView
  horizontal={true}
>
  <View>
    <View
      style={
        Object {
          "paddingBottom": 10,
          "paddingLeft": 30,
          "paddingRight": 5,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://www.creativindie.com/wp-content/uploads/2012/07/stock-image-site-pinterest-graphic.jpg",
            }
          }
          style={
            Object {
              "borderRadius": 4,
              "height": 190,
              "resizeMode": "contain",
              "width": 140,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#6200ee",
              "marginBottom": 5,
            }
          }
        >
          Book One
        </Text>
        <Text
          style={
            Object {
              "color": "#6200ee",
              "marginBottom": 5,
            }
          }
        >
          AnyAuthor
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingBottom": 10,
          "paddingLeft": 30,
          "paddingRight": 5,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://edit.org/photos/img/blog/t9i-edit-book-covers-online.jpg-840.jpg",
            }
          }
          style={
            Object {
              "borderRadius": 4,
              "height": 190,
              "resizeMode": "contain",
              "width": 140,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#6200ee",
              "marginBottom": 5,
            }
          }
        >
          Book Two
        </Text>
        <Text
          style={
            Object {
              "color": "#6200ee",
              "marginBottom": 5,
            }
          }
        >
          AnyAuthor
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingBottom": 10,
          "paddingLeft": 30,
          "paddingRight": 5,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://www.adobe.com/express/create/cover/media_178ebed46ae02d6f3284c7886e9b28c5bb9046a02.jpeg?width=400&format=jpeg&optimize=medium",
            }
          }
          style={
            Object {
              "borderRadius": 4,
              "height": 190,
              "resizeMode": "contain",
              "width": 140,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#6200ee",
              "marginBottom": 5,
            }
          }
        >
          Book Three
        </Text>
        <Text
          style={
            Object {
              "color": "#6200ee",
              "marginBottom": 5,
            }
          }
        >
          AnyAuthor
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
